packages/server/api/src/app/core/security/authz/project-authz-handler.ts

```diff
@@ -1,55 +1,56 @@
 import {
     ActivepiecesError,
-    assertNotNullOrUndefined,
     ErrorCode,
-    PrincipalType,
 } from '@activepieces/shared'
-import { FastifyRequest } from 'fastify'
 import { requestUtils } from '../../request/request-utils'
-import { BaseSecurityHandler } from '../security-handler'
-
-export class ProjectAuthzHandler extends BaseSecurityHandler {
-    private static readonly IGNORED_ROUTES = [
-        '/v1/admin/pieces',
-        '/v1/admin/platforms',
-        '/v1/app-credentials',
-        '/v1/authentication/switch-project',
-        '/v1/authentication/switch-platform',
-        '/v1/webhooks',
-        '/v1/webhooks/:flowId',
-        '/v1/webhooks/:flowId/test',
-        '/v1/webhooks/:flowId/sync',
-        // This works for both platform and project, we have to check this manually
-        '/v1/user-invitations',
-        '/v1/audit-events',
-    ]
-
-    protected canHandle(request: FastifyRequest): Promise<boolean> {
-        const routerPath = request.routeOptions.url
-        assertNotNullOrUndefined(routerPath, 'routerPath is undefined')
-        const requestMatches = !ProjectAuthzHandler.IGNORED_ROUTES.includes(
-            routerPath,
-        )
-        return Promise.resolve(requestMatches)
-    }
+import { BaseAuthzHandler } from '../security-handler'
+import { AuthenticatedFastifyRequest } from '../../../../../types/fastify'
+import { projectMemberService } from '../../../ee/projects/project-members/project-member.service'
+import { AuthorizationType, ProjectAuthorization } from '@activepieces/server-shared'
+
+export class ProjectAuthzHandler extends BaseAuthzHandler<ProjectAuthorizedRequest> {
```

Comment:
I think the authorization also can be simpler, one entry to the code.

export const securityAuthorization= {
    authorize()
}

and it read which type and validate based on that instead of classes and securty chain handler